### ISSUE TITLE
Variable font support

### ISSUE THREAD

**Issue by hansmbakker** [dotnet/maui#16604]
### Description

I would like to consume [variable fonts](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Fonts/Variable_Fonts_Guide) in .NET MAUI. 

### Public API Changes

It would be nice to have an overload of the current font loading API that takes a dictionary of variation settings to define labeled combinations.

```csharp
var variationSettingsA = new Dictionary<string, double>()
{
    { "wght", 700d},
    { "ital", 0d }
};

var variationSettingsB = new Dictionary<string, double>()
{
    { "wght", 300d},
    { "ital", 1d }
};

builder.ConfigureFonts(
            fonts =>
            {
                fonts.AddFont(filename: "myVariableFont.otf", label: "FontVariationA", variationSettings: variationSettingsA);
                fonts.AddFont(filename: "myVariableFont.otf", label: "FontVariationB", variationSettings: variationSettingsB);
            });
```

### Intended Use-Case

One of the things this is needed for, is to use the [Material Symbols](https://fonts.google.com/icons?icon.set=Material+Symbols), which are [shipped as variable fonts](https://github.com/google/material-design-icons/tree/master/variablefont).
Besides these Material Symbols, supporting variable fonts allows app developers to have more flexibility with their typography in general.

> Note: the fixed fonts in the linked repo are the legacy Material Icons, not the newer Material Symbols I want to use

**Comment by hansmbakker** [comment]
The manual workaround for this now seems to be using the [mutator tool from fontTools](https://graphicdesign.stackexchange.com/a/158774).

**Comment by mklts** [comment]
this feature is blocked by a single method missing in a SkiaSharp API https://github.com/mono/SkiaSharp/issues/2397

**Comment by eddieyanez** [comment]
> this feature is blocked by a single method missing in a SkiaSharp API [mono/SkiaSharp#2397](https://github.com/mono/SkiaSharp/issues/2397)

Does this mean that this can now be implemented in MAUI?

**Comment by beeradmoore** [comment]
@mklts , I don't follow. Wouldn't this feature rely on native `UIFont` and `Font` stuff from iOS and Android? Why would SkiaSharp be involved?
